aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,205 @@
+
+MVC will continue validating fields until reaches the maximum number of errors (200 by default). You can configure this number by inserting the following code into the ``ConfigureServices`` method in the ``Startup.cs`` file:
+
+.. literalinclude:: validation/sample/Startup.cs
+   :language: c#
+   :lines: 5-14
+   :dedent: 4
+
+Manual validation
+-----------------
+
+After model binding and validation are complete, you may want to repeat parts of it. For example, a user may have entered text in a field expecting an integer, or you may need to compute a value for a model's property. 
+
+You may need to run validation manually. To do so, call the ``TryValidateModel`` method, as shown here: 
+
+.. code-block:: c#
+
+  TryValidateModel(movie);
+   
+  
+Custom validation
+-----------------
+
+Validation attributes work for most validation needs. However, some validation rules are specific to your business, as they're not just generic data validation such as ensuring a field is required or that it conforms to a range of values. For these scenarios, custom validation attributes are a great solution. Creating your own custom validation attributes in MVC is easy. Just inherit from the ``ValidationAttribute``, and override the ``IsValid`` method. The ``IsValid`` method accepts two parameters, the first is an object named `value` and the second is a ``ValidationContext`` object named `validationContext`. `Value` refers to the actual value from the field that your custom validator is validating. 
+
+In the following sample, a business rule that states that users may not set the genre to `Classic` for a movie released after 1960. The ``[ClassicMovie]`` attribute checks the genre first, and if it is a classic, then it checks the release date to see that it is later than 1960. If it is released after 1960, validation fails. The attribute accepts an integer parameter representing the year that you can use to validate data. You can capture the value of the parameter in the attribute's constructor, as shown here:
+					 
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+   :language: c#
+   :lines: 11-37
+   :dedent: 4
+   
+The ``movie`` variable above represents a ``Movie`` object that contains the data from the form submission to validate. In this case, the validation code checks the date and genre in the ``IsValid`` method of the ``ClassicMovieAttribute`` class as per the rules. Upon successful validation ``IsValid`` returns a ``ValidationResult.Success`` code, and when validation fails, a ``ValidationResult`` with an error message. When a user modifies the ``Genre`` field and submits the form, the ``IsValid`` method of the ``ClassicMovieAttribute`` will verify whether the movie is a classic. Like any built-in attribute, apply the ``ClassicMovieAttribute`` to a property such as ``ReleaseDate`` to ensure validation happens, as shown in the previous code sample. 
```

Comment:
Hadn't noticed before (sorry) but this `ValidationAttribute` is oddly constrained. It'll work _only_ on a property of the `Movie` type.

Instead of changing the sample to create a new validation attribute, I suggest mentioning this is exactly one of those cases where `IValidatableObject` is more sensible.